docs/reference/index-modules.asciidoc

```diff
@@ -329,6 +329,13 @@ Defaults to `*`, which matches all fields eligible for
     the default pipeline (if it exists). The special pipeline name `_none`
     indicates no ingest pipeline will run.
 
+[[index-mapping-dimension-fields-limit]]
+`index.mapping.dimension_fields.limit`::
+For internal use by {kib} only. Maximum number of time series dimensions for the
```

Comment:
I don't understand how it relates to Kibana?

Reply:
We previously discussed marking the `dimension` parameter (and by extension this setting) for internal use by
the Observability (specifically metrics) solution only. That solution is shipped as a part of Kibana.

If that's changed, let me know. I can update this in a separate PR.

Reply:
Thanks for the context. My understanding is that this was indeed the initial vision, but that we moved from per-field metadata to dedicated mapping parameters with the vision that these parameters would be used by Elasticsearch in the long term as well. E.g. rollups could auto-configure based on what fields are marked as dimensions, and the timeseries_mode setting could be computed automatically based on dimensions. None of these have been merged yet but hopefully this dimension flag is going to be used by more than Kibana in the near future. @csoulios Please keep me honest if the above is incorrect.

Reply:
@jpountz your understanding is correct. `dimension` and `metric` mapping parameters will be used by ES (both for TSDB and rollups) for discovering dimensions and metrics. Eventually, they will be available to end-users too.

It's only that at this point we want to keep this feature for internal use only until we feel confident there will be no changes required. See comment: https://github.com/elastic/elasticsearch/pull/74450#pullrequestreview-691233131

Reply:
Thanks for the discussion. That aligns with my understanding. The important point is that end users should not be using the `dimension` parameter at this time.

I'll push a change to broaden the copy to "For internal use by Elastic only."